src/pudl/transform/ferc1.py

```diff
@@ -351,6 +355,79 @@ def align_row_numbers_dbf(
     return df
 
 
+class UnstackBalancesToReportYearInstantXbrl(TransformParams):
+    """Parameters for :func:`unstack_balances_to_report_year_instant_xbrl`."""
+
+    unstack_balances_to_report_year: bool = False
+    """If True unstack balances to a single year (the report year)."""
+
+
+def unstack_balances_to_report_year_instant_xbrl(
+    df: pd.DataFrame, params: UnstackBalancesToReportYearInstantXbrl
+) -> pd.DataFrame:
+    """Turn start year end year rows into columns for each value type.
+
+    This function is utilized in :func:`process_instant_xbrl`.
+
+    There are some instant tables that report the start-of-year data and the
+    end-of-year data on seperate rows. The dbf version of the table has a column for the
+    starting data, a column for the ending data, and a row number that cooresponds with
+    the row literal that data represents: i.e., cost, etc.
+
+    This function unstacks that table and adds the suffixes ``_starting_balance`` and
+    ``_ending_balance`` to each of the columns. These may then be used as
+    ``value_types`` in the :func:`wide_to_tody` function to normalize the table.
+
+    There are two checks in place:
+
+    First, it will make sure that there are not multiple entries per year for the same
+    entiity_id. Ex: a row for 2020-12-31 and 2020-06-30 for entitiy_id X means that
+    there's more data here than is getting reported. We could just drop these mid-year
+    values, but we might want to keep them or at least check that there is no funny
+    business with the data.
+
+    It will also check that there are no mid-year dates period. If an entity reports
+    a value from the middle of the year, there is no telling whether that actually
+    represents the start or end balance and requires a closer look.
+    """
+    if params is None:
+        params = UnstackBalancesToReportYearInstantXbrl()
+    if params.unstack_balances_to_report_year:
+        df["year"] = pd.to_datetime(df["date"]).dt.year
+        if df.duplicated(["entity_id", "year"]).any():
+            raise AssertionError(
+                "Looks like there are multiple entries per year--not sure which to use "
+                "for the start/end balance."
+            )
+        if not pd.to_datetime(df["date"]).dt.is_year_end.all():
+            raise AssertionError(
+                "Looks like there are some values in here that aren't from the end of "
+                "the year. We can't use those to calculate start and end balances."
+            )
```

Comment:
Not sure both of these are necessary? Might be able to get away with just the second one if the drop dupes is also implemented.